relay-general/src/store/transactions/rules.rs

```diff
@@ -0,0 +1,141 @@
+use chrono::{DateTime, Utc};
+use relay_common::Glob;
+use serde::{Deserialize, Serialize};
+
+use crate::protocol::TransactionSource;
+
+/// The rule describes how transaction name should be changed.
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct TransactionNameRule {
+    /// The pattern which will be applied to transaction name.
+    pub pattern: Glob,
+    /// Date time when the rule expires and it should not be applied anymore.
+    pub expiry: DateTime<Utc>,
+    /// Object containing transaction attributes the rules must only be applied to.
+    #[serde(default)]
```

Comment:
I'm was using our docs for rules definition format, and set all the defaults as the relay supposed to use. From my point of few it's better to get all of those set on deserialization rather than using `Option` and then check additionally if it's set and `unwrap` with some defaults. 

E.g. `RuleScope` which is optional completely, but we default to `url` - and it much more convenient to set it straight away to what we expect to be default. 

Let me know what you think.

Reply:
I personally would not set any defaults. If relay can't deserialize a rule because it doesn't know about it, then ignore the rule.